internal/distro/rhel84/distro_v2.go

```diff
@@ -448,3 +572,173 @@ func (t *imageTypeS2) systemdStageOptions(enabledServices, disabledServices []st
 		DefaultTarget:    target,
 	}
 }
+
+func (t *imageTypeS2) buildStampStageOptions() *osbuild.BuildstampStageOptions {
+	return &osbuild.BuildstampStageOptions{
+		Arch:    t.Arch().Name(),
+		Product: "Red Hat Enterprise Linux",
+		Version: "8.4",
+		Variant: "edge",
+		Final:   true,
+	}
+}
+
+func (t *imageTypeS2) anacondaStageOptions() *osbuild.AnacondaStageOptions {
+	return &osbuild.AnacondaStageOptions{
+		KickstartModules: []string{
+			"org.fedoraproject.Anaconda.Modules.Network",
+			"org.fedoraproject.Anaconda.Modules.Payloads",
+			"org.fedoraproject.Anaconda.Modules.Storage",
+		},
+	}
+}
+
+func (t *imageTypeS2) loraxScriptStageOptions() *osbuild.LoraxScriptStageOptions {
+	return &osbuild.LoraxScriptStageOptions{
+		Path:     "99-generic/runtime-postinstall.tmpl",
+		BaseArch: t.Arch().Name(),
+	}
+}
+
+func (t *imageTypeS2) dracutStageOptions(kernelVer string) *osbuild.DracutStageOptions {
+	kernel := []string{kernelVer}
+	modules := []string{
+		"bash",
+		"systemd",
+		"fips",
+		"systemd-initrd",
+		"modsign",
+		"nss-softokn",
+		"rdma",
+		"rngd",
+		"i18n",
+		"convertfs",
+		"network-manager",
+		"network",
+		"ifcfg",
+		"url-lib",
+		"drm",
+		"plymouth",
+		"prefixdevname",
+		"prefixdevname-tools",
+		"anaconda",
+		"crypt",
+		"dm",
+		"dmsquash-live",
+		"kernel-modules",
+		"kernel-modules-extra",
+		"kernel-network-modules",
+		"livenet",
+		"lvm",
+		"mdraid",
+		"multipath",
+		"qemu",
+		"qemu-net",
+		"fcoe",
+		"fcoe-uefi",
+		"iscsi",
+		"lunmask",
+		"nfs",
+		"resume",
+		"rootfs-block",
+		"terminfo",
+		"udev-rules",
+		"biosdevname",
+		"dracut-systemd",
+		"pollcdrom",
+		"usrmount",
+		"base",
+		"fs-lib",
+		"img-lib",
+		"shutdown",
+		"uefi-lib",
+	}
+	return &osbuild.DracutStageOptions{
+		Kernel:  kernel,
+		Modules: modules,
+		Install: []string{"/.buildstamp"},
+	}
+}
+
+func (t *imageTypeS2) kickstartStageOptions(ostreeURL string) *osbuild.KickstartStageOptions {
+	return &osbuild.KickstartStageOptions{
+		Path: "/usr/share/anaconda/interactive-defaults.ks",
+		OSTree: osbuild.OSTreeOptions{
+			OSName: "rhel",
+			URL:    ostreeURL,
+			Ref:    t.OSTreeRef(),
+			GPG:    false,
+		},
+	}
+}
+
+func (t *imageTypeS2) bootISOMonoStageOptions(kernelVer string) *osbuild.BootISOMonoStageOptions {
+	return &osbuild.BootISOMonoStageOptions{
+		Product: osbuild.Product{
+			Name:    "Red Hat Enterprise Linux",
+			Version: "8.4",
+		},
+		ISOLabel: "RHEL-8-4-X86_64",
```

Comment:
Should this be arch-independent?

Reply:
This is actually used to build the kernel command line `cmdline = f"inst.stage2=hd:LABEL={isolabel}"` which means it needs to match the `VolID` of the below, i.e. be `fmt.Sprintf("RHEL-8-4-0-BaseOS-%s", t.Arch().Name()),`.

Reply:
Fixed to match the VolID.  Thanks!